File: Practica 14/NodoBinario.py
class NodoBinario():
	##
	#Constructor
	##
	def __init__(self,dato):
		self.__dato = dato
		self.__left = None
		self.__right = None

	##
	#Gets
	##
	def getDato(self):
		return self.__dato

	def getLeft(self):
		return self.__left

	def getRight(self):
		return self.__right

	##
	#Sets
	##
	def setDato(self,where):
		self.__dato = where

	def setLeft(self,where):
		self.__left = where

	def setRight(self,where):
		self.__right = where

	##
	#Funciones
	##
	def mostrarPreOrden(self):
		print (self.getDato())
		if (self.__left != None):
			self.__left.mostrarPreOrden()
		if(self.__right != None):
			self.__right.mostrarPreOrden()

	def mostrarEnOrden(self):
		if(self.__left != None):
			self.__left.mostrarEnOrden()
		print (self.getDato())
		if(self.__right != None):
			self.__right.mostrarEnOrden()

	def mostrarPosOrden(self):
		if(self.__left != None):
			self.__left.mostrarPosOrden()
		if(self.__right != None):
			self.__right.mostrarPosOrden()
		print (self.getDato())

File: Practica 14/NodoBinario.java
public class NodoBinario{

	private int valor;
	private NodoBinario padre;
	private NodoBinario left;
	private NodoBinario right;

	/**
	 * Constructor
	 */
	public NodoBinario(int valor) {
		this.valor = valor;
	}

	/**
	 * Gets
	 */
	public int getValor(){
		return valor;
	}

	public NodoBinario getPadre(){
		return padre;
	}

	public NodoBinario getLeft(){
		return left;
	}

	public NodoBinario getRight(){
		return right;
	}

	/**
	 * Sets
	 */
	public void setValor(int valor){
		this.valor = valor;
	}

	public void setPadre(NodoBinario padre){
		this.padre = padre;
	}

	public void setLeft(NodoBinario left){
		this.left = left;
	}

	public void setRight(NodoBinario right){
		this.right = right;
	}

	/**
	 * Funciones
	 */
	public void mostrarPreOrden(){
		System.out.println (getValor());
		if (left != null)
			left.mostrarPreOrden();
		if(right != null)
			right.mostrarPreOrden();
	}

	public void mostrarEnOrden(){
		if(left != null)
			left.mostrarEnOrden();
		System.out.println (getValor());
		if(right != null)
			right.mostrarEnOrden();
	}

	public void mostrarPosOrden(){
		if(left != null)
			left.mostrarPosOrden();
		if(right != null)
			right.mostrarPosOrden();
		System.out.println (getValor());
	}
}
